### ISSUE TITLE
Remove `==`/`!=`/`=`/`<>` whose operands are bool value (true / false) and literal, or replace them with `!`/`Not` if necessary

### ISSUE THREAD

**Issue by tats-u** [dotnet/roslyn#72506]
**Brief description:**

Describe your **code style** rule here.

Newbies especially who have not got used to `!` or `Not` tend to write `== true`, `== false`, `= True`, etc., which are redundant.

This rule enforces users to write bool values as is or with `!` or `Not`, which are shorter and (VB only) readable as a more natural English sentence.

**Languages applicable:**

Is your analyzer C# only? VB Only? Or Both?

Both

**Code example that the analyzer should report:**

A small code snippet that describes a case that the analyzer should report.

```cs
if (string.IsNullOrEmpty(strInput) == true)
```

↓

```cs
if (string.IsNullOrEmpty(strInput))
```

```vbnet
If String.Contains("mandatory") = False Then
```

↓

```vbnet
If Not String.Contains("mandatory") Then
```

**Additional information:**

Any more additional information you would like to add.

This must not be applied to the nullable bool type.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by tats-u** [comment]
I overlooked it probably because it is disabled by default. Sorry.

**Comment by CyrusNajmabadi** [comment]
No worries at all :)

**Comment by tats-u** [comment]
I confirmed `== true` is warned by default, but `== false` is not by default. 
Has it really been implemented like `== true`? Is it correct it is suppressed by default?
